js/interface.js

```diff
@@ -4,265 +4,284 @@ function toggleFields(value = false) {
   $('#typeaheadUserName').closest('span').toggle(value);
 }
 
-function manageDataSourceChange(dataSourceId) {
-  if (dataSourceId) {
-    toggleFields(true);
 
-    return Fliplet.DataSources.getById(dataSourceId, {
-      attributes: ['columns']
-    }).then(function(columns) {
-      $('#columnEmail').html('');
-      $('#columnUserPhoto').html('');
-      $('#columnEmail').append('<option value="">Select an option</option>');
-      $('#columnUserPhoto').append(
-        '<option value="">Select an option</option>'
-      );
-      columns.columns.forEach((el) => {
-        $('#columnEmail').append(`<option value="${el}">${el}</option>`);
-        $('#columnUserPhoto').append(`<option value="${el}">${el}</option>`);
-      });
+Fliplet.Widget.findParents({ filter: { package: 'com.fliplet.dynamic-container' } }).then(function(widgets) {
+  const dynamicContainer = widgets[0];
 
-      var instance = Fliplet.UI.Typeahead($('#typeaheadUserName'), {
-        // options: columns.columns.map(el => {
-        //   return { value: el, label: el };
-        // }),
-        freeInput: false,
-        maxItems: 2
-      });
+  if (widgets.length === 0 || !dynamicContainer.dataSourceId) {
+    Fliplet.Widget.generateInterface({
+      title: 'Configure data comments',
+      fields: [
+        {
+          type: 'html',
+          html: '<p style="color: #A5A5A5; font-size: 12px; font-weight: 400;">This component needs to be placed inside a Data container with selected Data source</p>'
+        }
+      ]
+    });
 
-      instance.options(
-        columns.columns.map((el) => {
-          return { value: el, label: el };
-        })
-      );
+    return;
+  }
 
-      const key = Object.keys(__widgetData)[0];
-      const objValue = __widgetData[key].data;
+  function manageDataSourceChange(dataSourceId) {
+    if (dataSourceId) {
+      toggleFields(true);
 
-      instance.set(objValue.userNames, true);
-      $('#columnUserPhoto').val(objValue.columnUserPhoto);
-      $('#columnEmail').val(objValue.columnEmail);
+      return Fliplet.DataSources.getById(dataSourceId, {
```

Comment:
@armine-fliplet Is it possible to use try catch block here for readability purpose.